accelerator/core/mam_core.c

```diff
@@ -241,10 +246,14 @@ static status_t create_channel_fetch_all_transactions(const iota_client_service_
     goto done;
   }
 
-  if (mam_api_channel_create(api, channel_depth, chid) != RC_OK) {
-    ret = SC_MAM_FAILED_CREATE_OR_GET_ID;
-    ta_log_error("%s\n", ta_error_to_string(ret));
-    goto done;
+  // We have created a channel when we want to get to teh given channel at the first loop in
+  // `ta_mam_written_msg_to_bundle()`, so we don't need to create a channel once again.
+  if (!first_loop) {
```

Comment:
Can you determinate the condition if it is first loop or not without having to specify automatically?

Reply:
It is quite hard to find a solution at this moment

Reply:
Add FIXME

Reply:
What is the difference between first loop and first iteration?

Reply:
I should name it as first iteration of the loop instead of first loop